### ISSUE TITLE
Testcase 8B (XML Nuytinck) invalid case in Media reference 

### ISSUE THREAD

**Issue by sam-medispring** [hl7-be/lab#290]
Reference urls are Case sensitive and thus it should be `Media/a.pdf `

http://hl7.org/fhir/R4/references.html#regex

![image](https://github.com/hl7-be/lab/assets/70625381/74adf9fd-7174-4f32-b527-dd88336e9c7f)

**Comment by bdc-ehealth** [comment]
@sam-medispring ,

Indeed, the case used here is different from the other elements in the file. However, since the use of "fullUrl" by the validator has changed anyway, it would be better not to rely on this.  The recommended approach in a bundle that requires a fullUrl is now to use a "urn:uuid" to refer to the included resources.

**Comment by sam-medispring** [comment]
@bdc-ehealth I'm not sure i understand. 
As a receiver (GP soft): we cannot handle this file because the reference doesn't point to anything. We don't use the fullUrl. We use the reference to find the enty that matches (Media / Id='...')

**Comment by bdc-ehealth** [comment]
@sam-medispring ,

The rules for resolution in a bundle are described here: https://www.hl7.org/fhir/R4/bundle.html#bundle-unique . I have read them again, but I cannot find a place where it says that casing impacts the resolution. But maybe I am wrong. Can you indicate where you see this? 

For an explanation of my comment above, see: https://www.hl7.org/fhir/R4/bundle.html#:~:text=Where%20a%20resource%20is%20not%20assigned%20a%20persistent%20identity%20that%20can%20be%20used%20in%20the%20Bundle%2C%20a%20UUID%20should%20be%20used%20(urn%3Auuid%3A...).

**Comment by LotuxPunk** [comment]
@bdc-ehealth

Hello Bart,
It is indicated that "URLs are always considered to be case-sensitive" in the documentation when you click on the "(see the RESTful URL regex)." (https://www.hl7.org/fhir/R4/references.html#regex)

For the rest, I agree with you, if the references across a bundle are urn:oid or urn:uuid the problem would be solved. However, in the case of a relative URL, it must follow the regex and the attached rules.

**Comment by bdc-ehealth** [comment]
@LotuxPunk , @sam-medispring ,

I agree that URLs are case sensitive, but this relative URL "media/a.pdf" does not resolve to an absolute URL (this resource is not persisted on a FHIR server). So I ask myself, what does the "media" in the relative URL refer to? Not to the element (tagname) "Media" with a capital letter, because in that case it would be an xpath or something like that, not an URL. You can only resolve based on the id "a.pdf"... so it does not matter in this case.

**Comment by LotuxPunk** [comment]
Hello @bdc-ehealth ,
I do not agree with you. If you read the spec (https://build.fhir.org/bundle-definitions.html#Bundle.entry.fullUrl), the `fullUrl` of an entry MUST be an Absolute Url (in this case, the bundle is a `document`, so the exceptions do not apply).
Consequently, `media/a.pdf` does not correspond to an absolute URL and is therefore invalid as far as I understand the spec.

**Comment by bdc-ehealth** [comment]
@LotuxPunk ,

Indeed, "media/a.pdf" is invalid, but it should be replaced by an "urn:uuid:"-style URL, not by "Media/a.pdf".

**Comment by AlexisVZ-MDS** [comment]
OK, so you will ask labo Nuytinck to provide a new test file ?

**Comment by bdc-ehealth** [comment]
@AlexisVZ-MDS ,

Alexis, it is not necessary to ask for a new file, because we accept the "invalid" URL for the time being. For your purposes of analyzing and displaying, you can use the bare "id" (others GP softwares did it that way). If you want to talk about it, we can have a phonecall.

**Comment by AlexisVZ-MDS** [comment]
@LotuxPunk can you please have a look to implement the workaround of using the "id" ? 
So, I understand it now:
![image](https://github.com/hl7-be/lab/assets/93183816/8ca37b61-7ac3-4a32-8367-4a58f4619e86)
